app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/HomeScreenShimmerLoading.kt

```diff
@@ -0,0 +1,134 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+
+@Composable
+fun HomeScreenShimmerLoading() {
+    val shimmerColors = listOf(
+        Color.LightGray.copy(alpha = 0.4f),
+        Color.LightGray.copy(alpha = 0.2f),
+        Color.LightGray.copy(alpha = 0.4f),
+    )
+
+    val transition = rememberInfiniteTransition()
+    val translateAnimation = transition.animateFloat(
+        initialValue = 0f,
+        targetValue = 1000f,
+        animationSpec = infiniteRepeatable(
+            animation = tween(
+                durationMillis = 1000,
+                easing = FastOutSlowInEasing
+            ),
+            repeatMode = RepeatMode.Reverse
+        )
+    )
+
+    val brush = Brush.linearGradient(
+        colors = shimmerColors,
+        start = Offset.Zero,
+        end = Offset(x = translateAnimation.value, y = translateAnimation.value)
+    )
+
+    ShimmerHomeScreen(brush = brush)
+}
+
+@Composable
+fun ShimmerHomeScreen(brush: Brush) {
+    ConstraintLayout(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+    ) {
+
+        val (topTexts, bottomTexts, circle) = createRefs()
+
+        Spacer(
+            modifier = Modifier
+                .size(36.dp)
+                .clip(CircleShape)
+                .background(brush)
+                .constrainAs(circle) {
+                    top.linkTo(parent.top, 79.dp)
+                    end.linkTo(parent.end, 20.dp)
+                }
+        )
+        Column(
+            Modifier.constrainAs(topTexts) {
+                top.linkTo(parent.top, 61.dp)
+                start.linkTo(parent.start, 20.dp)
+            }
+        ) {
+            CustomSpacerRectangle(brush = brush, width = 117.dp)
+            CustomSpacerRectangle(brush = brush, width = 80.dp, topPadding = 15.dp)
+        }
+
+        Column(
+            Modifier.constrainAs(bottomTexts) {
+                bottom.linkTo(parent.bottom, 62.dp)
+                start.linkTo(parent.start, 20.dp)
+            }
+        ) {
+            CustomSpacerRectangle(brush = brush, width = 42.dp)
+            CustomSpacerRectangle(brush = brush, width = 253.dp, topPadding = 15.dp)
+            CustomSpacerRectangle(brush = brush, width = 126.dp, topPadding = 10.dp)
+            CustomSpacerRectangle(brush = brush, width = 318.dp, topPadding = 15.dp)
+            CustomSpacerRectangle(brush = brush, width = 170.dp, topPadding = 10.dp)
+        }
+    }
+}
+
+@Composable
+fun CustomSpacerRectangle(brush: Brush, width: Dp, topPadding: Dp = 0.dp) {
+    Column {
+        Spacer(modifier = Modifier.height(topPadding))
+        Spacer(
+            modifier = Modifier
+                .size(width = width, height = 20.dp)
+                .clip(RoundedCornerShape(10.dp))
+                .background(brush)
+        )
+    }
+}
```

Comment:
Why don't we pass in a `Modifier` instead of `topPadding`? That way it's more aligned with how Compose is written and we can simplify it 🤔 
```suggestion
fun HomeScreenShimmer(
    brush: Brush, 
    width: Dp, 
    modifier: Modifier = Modifier
) = Spacer(
    modifier = modifier
        .size(width = width, height = 20.dp)
        .clip(CircleShape)
        .background(brush)
)
```

Reply:
I am drawing those rectangles inside the `Column` and the space between each rectangle is not the same in the Figma. So I have to create a fake space with `Spacer`. Rectangles have different width and top padding too. Since this is just for the animation and don't really have much function, I decided to make it this way.

Reply:
That's why we pass in a `Modifier`, so that we can set the topPadding on the single `Spacer` in the composable. This way we only need to have one `Spacer` in our hierarchy per shimmer. For example:
```kotlin
val brush = Brush()
Column {
    HomeScreenShimmer(brush = brush, width = 50.dp, modifier = Modifier.paddingTop(8.dp))
    HomeScreenShimmer(brush = brush, width = 20.dp, modifier = Modifier.paddingTop(16.dp))
    HomeScreenShimmer(brush = brush, width = 70.dp, modifier = Modifier.paddingTop(8.dp))
}
```
The hierarchy for this will then be:
```
Column
    Spacer
    Spacer
    Spacer
```
instead of this for the current implementation:
```
Column
    Column
        Spacer
        Spacer
    Column
        Spacer
        Spacer
    Column
        Spacer
        Spacer
```